backend/src/main/java/com/google/backend/servlets/AlertsDataServlet.java

```diff
@@ -49,11 +54,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // TODO: only get alerts that the user is subscribed to.
     Query query = new Query(Alert.ALERT_ENTITY_KIND);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    List<Alert> alertList = new ArrayList<>();
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(MAX_LIMIT));    
+    List<Alert> alertList = results.stream()
+      .map(Alert::createAlertFromEntity).collect(Collectors.toList());
+    Collections.sort(alertList, Collections.reverseOrder());
 
-    for (Entity entity : results.asIterable()) {
-      alertList.add(Alert.createAlertFromEntity(entity));
+    int limit = Integer.parseInt(request.getParameter("limit"));
```

Comment:
nit: if client side sends a non-parsable parameter (i know it won't since you wrote the client, but in case that someone else wrote the client...), maybe you can catch and log/deal with `NumberFormatException` if it is thrown.